src/main/java/io/reactivesocket/cli/Main.java

```diff
@@ -241,6 +317,26 @@ private Completable runSingleOperation(ReactiveSocket client) {
         return ObservableIO.lines(is);
     }
 
+    private String getMetadata() {
+        synchronized (this) {
+            if (cachedMetadata == null) {
+                if (metadata == null) {
+                    cachedMetadata = "";
+                } else if (metadata.startsWith("@")) {
```

Comment:
the overall block is slightly different as you don't use stdin.  But I'll refactor later if you want to punt the partial cleanup for now.  I'd argue yes, like clean readable code is good in tests as well as main code.